### ISSUE TITLE
Can't record audio on app targeting Android 13 due to plugin requesting WRITE_EXTERNAL_STORAGE permission (which always fails)

### ISSUE THREAD

**Issue by suzukieng** [apache/cordova-plugin-media#373]
# Bug Report

### What is expected to happen?

Recording audio should work on Android 13.

### What does actually happen?

Recording audio fails, because the permission request for `WRITE_EXTERNAL_STORAGE` always fails on Android 13.
The check is always for both `RECORD_AUDIO` and `WRITE_EXTERNAL_STORAGE`.

## Information

The app does not request legacy external storage.
I've seen #268 but from my understanding that change only removes the permission from the manifest, but it still tries to acquire it a runtime?

### Command or Code

`media.startRecord()`

### Environment, Platform, Device

Android 13 running on Samsung Galaxy S20

### Version information

cordova-media-plugin 6.1

## Checklist

- [x] I searched for existing GitHub issues
- [x] I updated all Cordova tooling to most recent version
- [x] I included all the necessary information above

**Comment by mirko77** [comment]
That was removed a while ago https://github.com/apache/cordova-plugin-media/commit/e8ca2244d04b5f9463b9c01ba1a47bce3eace0bb

**Comment by suzukieng** [comment]
> `WRITE_EXTERNAL_STORAGE` was removed a while ago [e8ca224](https://github.com/apache/cordova-plugin-media/commit/e8ca2244d04b5f9463b9c01ba1a47bce3eace0bb)

In the manifest, yes. But it’s still requested at run-time:

https://github.com/apache/cordova-plugin-media/blob/a8bbc02a06cbf01328738ee3e76cac7e32efa042/src/android/AudioHandler.java#L571

**Comment by mirko77** [comment]
Good point. That might explain why we cannot get it to work on Android 13 when targeting API 33, as that permission `WRITE_EXTERNAL_STORAGE ` will always be denied.

**Comment by mirko77** [comment]
I tried and it works, at least on Android 13 and 11 with API 33. I do not have any Android 12 devices.

`READ_MEDIA_AUDIO` is not needed.

@suzukieng you can try this fork https://github.com/epicollect5/cordova-plugin-media
@breautek I added a pull request

**Comment by suzukieng** [comment]
@mirko77 correct, `READ_MEDIA_AUDIO` was introduced in API level 33+. An app that previously used `READ_EXTERNAL_STORAGE` should now request `READ_MEDIA_*`.

So something like this is needed:
```
String READ_IMAGES_PERMISSION =
            Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU ?
                    Manifest.permission.READ_MEDIA_IMAGES : Manifest.permission.READ_EXTERNAL_STORAGE;
```

**Comment by mirko77** [comment]
@suzukieng have you tried my fork?

**Comment by adrianhardy** [comment]
Forgive my ignorance, but when looking at your suggested fix: 

```
String READ_IMAGES_PERMISSION =
            Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU ?
                    Manifest.permission.READ_MEDIA_IMAGES : Manifest.permission.READ_EXTERNAL_STORAGE;
```

Is that providing support for building using older "compileSDK"? Or is that a requirement to support multiple android versions at runtime?

Again, apologies if that's a silly question - I don't know a lot about Java, Android SDK, etc.

**Comment by suzukieng** [comment]
Hi @adrianhardy. It's just basically saying "if you're running on Android 13, request READ_MEDIA_IMAGES permission, otherwise request READ_EXTERNAL_STORAGE permission". So this makes an app targeting Android 13 comply with platform requirements, while falling back to legacy behavior on older Android versions (which have no READ_MEDIA_IMAGES permission). The ternary expression might make it a big hard to read.
